app/presenters/facet_presenter.rb

```diff
@@ -0,0 +1,181 @@
+  end
+
+  def basic_facet_item(item)
+    {
+      url: facet_item_url(item),
+      text: facet_label(@facet.name, item.value),
+      num_results: number_with_delimiter(item.hits),
+      is_checked: facet_in_params?(@facet.name, item)
+    }
+  end
+
+  def simple_facet
+    basic_facet(type: :simple)
+  end
+
+  def simple_facet_item(item)
+    basic_facet_item(item)
+  end
+
+  def colour_facet
+    basic_facet(type: :colour, count: 16)
+  end
+
+  def colour_facet_item(item)
+    basic_facet_item(item).tap do |basic|
+      basic.delete(:text)
+      basic[:hex] = item.value
+    end
+  end
+
+  def hierarchical_facet
+    basic_facet(type: :hierarchical)
+  end
+
+  def hierarchical_facet_item(item)
+    child_facets = hierarchical_facet_item_children(item)
+
+    {
+      has_subfilters: child_facets.present?,
+      filters: child_facets.map { |child|
+        self.class.new(child, @controller).display(true)
+      }
+    }.merge(simple_facet_item(item))
+  end
+
+  def hierarchical_facet_item_children(item)
+    facets_from_request(facet_field_names).select do |child|
+      parent = @configuration.facet_fields[child.name].parent
+      !parent.nil? && (parent.first == @facet.name) && (parent.last == item.value)
+    end
+  end
+
+  def range_facet
+    range_min = @facet.items.map(&:value).min
+    range_max = @facet.items.map(&:value).max
+    hits_max = @facet.items.map(&:hits).max
+    {
+      date: true,
+      title: facet_label(@facet.name),
+      form: {
+        action_url: search_action_url,
+        hidden_inputs: hidden_inputs_for_search
+      },
+      range: {
+        start: {
+          input_name: "range[#{facet.name}][begin]",
+          input_value: range_min,
+          label_text: 'From:'
+        },
+        end: {
+          input_name: "range[#{facet.name}][end]",
+          input_value: range_max,
+          label_text: 'To:'
+        }
+      },
+      data: @facet.items.sort_by(&:value).map do |item|
+        p = reset_search_params(params).deep_dup
+        p[:f] ||= {}
+        p[:f][@facet.name] = [item.value]
+        {
+          percent_of_max: (item.hits.to_f / hits_max.to_f * 100).to_i,
+          value: "#{item.value} (#{item.hits})",
+          url: search_action_path(p)
+        }
+      end,
+      date_start: range_min,
+      date_end: range_max
+    }
+  end
+
+  def boolean_facet
+    {
+      url: boolean_facet_url,
+      text: facet_label(@facet.name),
+      is_checked: boolean_facet_checked?
+    }
+  end
+
+  def boolean_facet_url
+    if boolean_facet_in_params?(@facet.name)
+      search_action_url(remove_facet_params(@facet.name, facet_params(@facet.name).first, params))
+    elsif boolean_facet_checked?
+      search_action_url(add_facet_params_and_redirect(@facet.name, facet_config.boolean[:off]))
+    else
+      search_action_url(add_facet_params_and_redirect(@facet.name, facet_config.boolean[:on]))
+    end
+  end
+
+  def boolean_facet_checked?
```

Comment:
Assignment Branch Condition size for boolean_facet_checked? is too high. [21.38/15]